core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -92,16 +93,17 @@ final class HttpChannelPool implements AsyncCloseable {
     // Fields for pooling connections:
     private final Map<PoolKey, Deque<PooledChannel>>[] pool;
     private final Map<PoolKey, ChannelAcquisitionFuture>[] pendingAcquisitions;
+    private final Map<Future<Channel>, ConnectionEventState> connectionEventStates;
```

Comment:
I agree that the Future<Channel> is not a good key and passing it as an argument is much cleaner.
However,  since the `ConnectionEventState` is created inside the `sessionPromise` callback, there is no way to pass it to the `notifyConnect` method.